### PR TITLE
fix: too many portals

### LINKED ISSUE #22208: Large HogQL table insights (by number of rows) with datetime fields cause excessive DOM size and page crashes
## Bug description

In the last ~3 weeks (or so!), the PostHog website has been responding very slowly compared to previously. Navigating between screens using the left tab bar can take >30 seconds at times, as well as issues with insights loading (which I've seen reported in other Issues).

Today, the website is crashing entirely and displaying Chrome's "Aw snap!" `Error code: 5` screen.

Separately, queries that used to take <5 seconds to run now take >40 seconds in some cases.

## How to reproduce

1. Load `https://us.posthog.com/project/XXXXX`
2. Wait for it to load
3. Try clicking any other screen and wait for it to load very slowly

## Additional context
Screenshot 2024-05-09 at 11 58 54

### Debug info

```shell
Session: https://us.posthog.com/project/sTMFPsFhdP1Ssg/replay/018f5b0a-d3ac-7cc6-b56a-74e71d86962c?t=523 
Admin: http://go/adminOrgUS/018a1aa0-fba1-0000-7c63-3525ad2656c8 (Project: 34169)
Sentry: http://go/sentryUS/34169
```

### PR DESCRIPTION
**Pull request by pauldambra**
fixes https://github.com/PostHog/posthog/issues/22208

We're adding a DOM node for every `Popover` in the DOM. With enough present the browser gets upset. No bueno.

[The docs for floating-ui say](https://floating-ui.com/docs/FloatingPortal#:~:text=Note,not%20in%20use.):

> The portal component should be conditionally rendered based on the isOpen or mounted state (as seen above), rather than always rendered. This prevents a portal container from being mounted to the DOM when not in use.

So, let's do that.

|before|after|
|--|--|
|Screenshot 2024-06-09 at 23 19 39|Screenshot 2024-06-09 at 23 20 04|

### CODE REVIEW

File: frontend/src/lib/components/TZLabel/TZLabel.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    return <TZLabel time={now().subtract(2, 'day')} />
+}
+
+export function MoreThanADayAgoWithPopover(): JSX.Element {
+    return <TZLabel time={now().subtract(2, 'day')} showPopover={false} />
+}
+
+export function WithMoreThanOne(): JSX.Element {
+    /**
+     * really exists as a regression test...
```
Comment (daibhin) [inline frontend/src/lib/components/TZLabel/TZLabel.stories.tsx:33]: Should we write is as an integration test in that case?

**Review by daibhin — APPROVED** [review]
Fixes the main issue so good to go but one comment how it's tested

File: frontend/src/lib/components/TZLabel/TZLabel.stories.tsx (hunk 1 above)
Comment (pauldambra) [reply frontend/src/lib/components/TZLabel/TZLabel.stories.tsx:33]: yeah, lazy of me 🙈

Comment (pauldambra) [reply frontend/src/lib/components/TZLabel/TZLabel.stories.tsx:33]: (you're right, but I'm going to optimize for progress over "perfection" 🤣 )

**Pull request merged**